[2024-12-24T06:37:53.234+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-24T06:37:53.246+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: dag_Calendar_NTBK.NB_Notebook_job manual__2024-12-24T06:37:51.454895+00:00 [queued]>
[2024-12-24T06:37:53.250+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: dag_Calendar_NTBK.NB_Notebook_job manual__2024-12-24T06:37:51.454895+00:00 [queued]>
[2024-12-24T06:37:53.251+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-24T06:37:53.257+0000] {taskinstance.py:2889} INFO - Executing <Task(DockerOperator): NB_Notebook_job> on 2024-12-24 06:37:51.454895+00:00
[2024-12-24T06:37:53.260+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=1190) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-12-24T06:37:53.261+0000] {standard_task_runner.py:72} INFO - Started process 1192 to run task
[2024-12-24T06:37:53.265+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'dag_Calendar_NTBK', 'NB_Notebook_job', 'manual__2024-12-24T06:37:51.454895+00:00', '--job-id', '358', '--raw', '--subdir', 'DAGS_FOLDER/dag_calendar.py', '--cfg-path', '/tmp/tmpp6fvilwa']
[2024-12-24T06:37:53.267+0000] {standard_task_runner.py:105} INFO - Job 358: Subtask NB_Notebook_job
[2024-12-24T06:37:53.308+0000] {task_command.py:467} INFO - Running <TaskInstance: dag_Calendar_NTBK.NB_Notebook_job manual__2024-12-24T06:37:51.454895+00:00 [running]> on host 1673b9432498
[2024-12-24T06:37:53.370+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='dag_Calendar_NTBK' AIRFLOW_CTX_TASK_ID='NB_Notebook_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-24T06:37:51.454895+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-24T06:37:51.454895+00:00'
[2024-12-24T06:37:53.373+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-24T06:37:53.513+0000] {docker.py:379} INFO - Starting docker container from image spark-cluster:version-1.0.0
[2024-12-24T06:37:55.160+0000] {docker.py:73} INFO - Input Notebook:  /home/jovyan/Notebooks/NB_Calendar.ipynb
[2024-12-24T06:37:55.165+0000] {docker.py:73} INFO - Output Notebook: /home/jovyan/Notebooks/output/NB_Calendar.ipynb
[2024-12-24T06:37:56.789+0000] {docker.py:73} INFO - Executing notebook with kernel: python3
[2024-12-24T06:37:56.794+0000] {docker.py:73} INFO - Executing Cell 1---------------------------------------
[2024-12-24T06:37:57.544+0000] {docker.py:73} INFO - Ending Cell 1------------------------------------------
[2024-12-24T06:37:57.551+0000] {docker.py:73} INFO - Executing Cell 2---------------------------------------
[2024-12-24T06:37:57.573+0000] {docker.py:73} INFO - Ending Cell 2------------------------------------------
[2024-12-24T06:37:57.578+0000] {docker.py:73} INFO - Executing Cell 3---------------------------------------
[2024-12-24T06:37:59.495+0000] {docker.py:73} INFO - :: loading settings :: url = jar:file:/usr/local/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-12-24T06:37:59.554+0000] {docker.py:73} INFO - Ivy Default Cache set to: /home/jovyan/.ivy2/cache
[2024-12-24T06:37:59.555+0000] {docker.py:73} INFO - The jars for the packages stored in: /home/jovyan/.ivy2/jars
[2024-12-24T06:37:59.557+0000] {docker.py:73} INFO - io.delta#delta-spark_2.12 added as a dependency
[2024-12-24T06:37:59.561+0000] {docker.py:73} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-5c11f010-b305-4fb6-a130-6db8d246cc87;1.0
[2024-12-24T06:37:59.562+0000] {docker.py:73} INFO - 	confs: [default]
[2024-12-24T06:38:10.181+0000] {docker.py:73} INFO - 	found io.delta#delta-spark_2.12;3.1.0 in central
[2024-12-24T06:38:10.191+0000] {docker.py:73} INFO - downloading https://repo1.maven.org/maven2/io/delta/delta-spark_2.12/3.1.0/delta-spark_2.12-3.1.0.jar ...
[2024-12-24T06:39:06.489+0000] {docker.py:73} INFO - 	[SUCCESSFUL ] io.delta#delta-spark_2.12;3.1.0!delta-spark_2.12.jar (56293ms)
[2024-12-24T06:39:06.494+0000] {docker.py:73} INFO - :: resolution report :: resolve 10632ms :: artifacts dl 56301ms
[2024-12-24T06:39:06.495+0000] {docker.py:73} INFO - 	:: modules in use:
[2024-12-24T06:39:06.495+0000] {docker.py:73} INFO - 	io.delta#delta-spark_2.12;3.1.0 from central in [default]
[2024-12-24T06:39:06.496+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-24T06:39:06.496+0000] {docker.py:73} INFO - 	|                  |            modules            ||   artifacts   |
[2024-12-24T06:39:06.497+0000] {docker.py:73} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2024-12-24T06:39:06.497+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-24T06:39:06.498+0000] {docker.py:73} INFO - 	|      default     |   1   |   1   |   0   |   0   ||   1   |   1   |
[2024-12-24T06:39:06.499+0000] {docker.py:73} INFO - 	---------------------------------------------------------------------
[2024-12-24T06:39:06.510+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:06.511+0000] {docker.py:73} INFO - :: problems summary ::
[2024-12-24T06:39:06.512+0000] {docker.py:73} INFO - :::: ERRORS
[2024-12-24T06:39:06.512+0000] {docker.py:73} INFO - 	Server access error at url https://repo1.maven.org/maven2/io/delta/delta-spark_2.12/3.1.0/delta-spark_2.12-3.1.0.pom (java.net.ConnectException: Connection refused)
[2024-12-24T06:39:06.513+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:06.514+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:06.514+0000] {docker.py:73} INFO - :: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
[2024-12-24T06:39:06.515+0000] {docker.py:73} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-5c11f010-b305-4fb6-a130-6db8d246cc87
[2024-12-24T06:39:06.516+0000] {docker.py:73} INFO - 	confs: [default]
[2024-12-24T06:39:06.537+0000] {docker.py:73} INFO - 	1 artifacts copied, 0 already retrieved (5374kB/26ms)
[2024-12-24T06:39:06.744+0000] {docker.py:73} INFO - 24/12/24 06:39:06 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-24T06:39:06.903+0000] {docker.py:73} INFO - Setting default log level to "WARN".
[2024-12-24T06:39:06.903+0000] {docker.py:73} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-12-24T06:39:07.508+0000] {docker.py:73} INFO - 24/12/24 06:39:07 WARN StandaloneSchedulerBackend: Dynamic allocation enabled without spark.executor.cores explicitly set, you may get more executors allocated than expected. It's recommended to set spark.executor.cores explicitly. Please check SPARK-30299 for more details.
[2024-12-24T06:39:07.849+0000] {docker.py:73} INFO - 24/12/24 06:39:07 WARN Utils: spark.executor.instances less than spark.dynamicAllocation.minExecutors is invalid, ignoring its setting, please update your configs.
[2024-12-24T06:39:08.527+0000] {docker.py:73} INFO - Driver Python Version: sys.version_info(major=3, minor=11, micro=6, releaselevel='final', serial=0)
[2024-12-24T06:39:08.529+0000] {docker.py:73} INFO - Worker Python Version: /opt/conda/bin/python
[2024-12-24T06:39:08.819+0000] {docker.py:73} INFO - <pyspark.sql.session.SparkSession at 0xffff65320450>
[2024-12-24T06:39:08.823+0000] {docker.py:73} INFO - Ending Cell 3------------------------------------------
[2024-12-24T06:39:08.825+0000] {docker.py:73} INFO - Executing Cell 4---------------------------------------
[2024-12-24T06:39:08.841+0000] {docker.py:73} INFO - Ending Cell 4------------------------------------------
[2024-12-24T06:39:08.843+0000] {docker.py:73} INFO - Executing Cell 5---------------------------------------
[2024-12-24T06:39:08.851+0000] {docker.py:73} INFO - Ending Cell 5------------------------------------------
[2024-12-24T06:39:08.853+0000] {docker.py:73} INFO - Executing Cell 6---------------------------------------
[2024-12-24T06:39:09.729+0000] {docker.py:73} INFO - Ending Cell 6------------------------------------------
[2024-12-24T06:39:09.731+0000] {docker.py:73} INFO - Executing Cell 7---------------------------------------
[2024-12-24T06:39:09.811+0000] {docker.py:73} INFO - Ending Cell 7------------------------------------------
[2024-12-24T06:39:09.813+0000] {docker.py:73} INFO - Executing Cell 8---------------------------------------
[2024-12-24T06:39:09.962+0000] {docker.py:73} INFO - Ending Cell 8------------------------------------------
[2024-12-24T06:39:09.963+0000] {docker.py:73} INFO - Executing Cell 9---------------------------------------
[2024-12-24T06:39:11.650+0000] {docker.py:73} INFO - [Stage 0:>                                                          (0 + 2) / 2]
[2024-12-24T06:39:13.060+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:13.082+0000] {docker.py:73} INFO - 18628
[2024-12-24T06:39:13.083+0000] {docker.py:73} INFO - Ending Cell 9------------------------------------------
[2024-12-24T06:39:13.090+0000] {docker.py:73} INFO - Executing Cell 10--------------------------------------
[2024-12-24T06:39:13.150+0000] {docker.py:73} INFO - Ending Cell 10-----------------------------------------
[2024-12-24T06:39:13.152+0000] {docker.py:73} INFO - Executing Cell 11--------------------------------------
[2024-12-24T06:39:13.637+0000] {docker.py:73} INFO - Ending Cell 11-----------------------------------------
[2024-12-24T06:39:16.299+0000] {docker.py:73} INFO - Traceback (most recent call last):
[2024-12-24T06:39:16.305+0000] {docker.py:73} INFO -   File "/opt/conda/bin/papermill", line 8, in <module>
[2024-12-24T06:39:16.305+0000] {docker.py:73} INFO -     sys.exit(papermill())
[2024-12-24T06:39:16.305+0000] {docker.py:73} INFO -              ^^^^^^^^^^^
[2024-12-24T06:39:16.305+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
[2024-12-24T06:39:16.306+0000] {docker.py:73} INFO -     return self.main(*args, **kwargs)
[2024-12-24T06:39:16.306+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-24T06:39:16.306+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1078, in main
[2024-12-24T06:39:16.306+0000] {docker.py:73} INFO -     rv = self.invoke(ctx)
[2024-12-24T06:39:16.306+0000] {docker.py:73} INFO -          ^^^^^^^^^^^^^^^^
[2024-12-24T06:39:16.307+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
[2024-12-24T06:39:16.307+0000] {docker.py:73} INFO -     return ctx.invoke(self.callback, **ctx.params)
[2024-12-24T06:39:16.307+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-24T06:39:16.307+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/core.py", line 783, in invoke
[2024-12-24T06:39:16.307+0000] {docker.py:73} INFO -     return __callback(*args, **kwargs)
[2024-12-24T06:39:16.307+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-24T06:39:16.308+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/click/decorators.py", line 33, in new_func
[2024-12-24T06:39:16.308+0000] {docker.py:73} INFO -     return f(get_current_context(), *args, **kwargs)
[2024-12-24T06:39:16.308+0000] {docker.py:73} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-24T06:39:16.308+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/cli.py", line 235, in papermill
[2024-12-24T06:39:16.308+0000] {docker.py:73} INFO -     execute_notebook(
[2024-12-24T06:39:16.308+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/execute.py", line 131, in execute_notebook
[2024-12-24T06:39:16.309+0000] {docker.py:73} INFO -     raise_for_execution_errors(nb, output_path)
[2024-12-24T06:39:16.309+0000] {docker.py:73} INFO -   File "/opt/conda/lib/python3.11/site-packages/papermill/execute.py", line 251, in raise_for_execution_errors
[2024-12-24T06:39:16.309+0000] {docker.py:73} INFO -     raise error
[2024-12-24T06:39:16.310+0000] {docker.py:73} INFO - papermill.exceptions.PapermillExecutionError: 
[2024-12-24T06:39:16.311+0000] {docker.py:73} INFO - ---------------------------------------------------------------------------
[2024-12-24T06:39:16.311+0000] {docker.py:73} INFO - Exception encountered at "In [11]":
[2024-12-24T06:39:16.311+0000] {docker.py:73} INFO - ---------------------------------------------------------------------------
[2024-12-24T06:39:16.311+0000] {docker.py:73} INFO - Py4JJavaError                             Traceback (most recent call last)
[2024-12-24T06:39:16.311+0000] {docker.py:73} INFO - Cell In[11], line 1
[2024-12-24T06:39:16.312+0000] {docker.py:73} INFO - ----> 1 if DeltaTable.isDeltaTable(spark, trgt_path_processed):
[2024-12-24T06:39:16.312+0000] {docker.py:73} INFO -       2     column_name = df_output.columns
[2024-12-24T06:39:16.312+0000] {docker.py:73} INFO -       3     set_clause = ", ".join([f"target.{i} = source.{i}" for i in column_name])
[2024-12-24T06:39:16.312+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.312+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/delta/tables.py:562, in DeltaTable.isDeltaTable(cls, sparkSession, identifier)
[2024-12-24T06:39:16.312+0000] {docker.py:73} INFO -     559 jvm: "JVMView" = sparkSession._sc._jvm  # type: ignore[attr-defined]
[2024-12-24T06:39:16.313+0000] {docker.py:73} INFO -     560 jsparkSession: "JavaObject" = sparkSession._jsparkSession  # type: ignore[attr-defined]
[2024-12-24T06:39:16.313+0000] {docker.py:73} INFO - --> 562 return jvm.io.delta.tables.DeltaTable.isDeltaTable(jsparkSession, identifier)
[2024-12-24T06:39:16.313+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.313+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/py4j/java_gateway.py:1322, in JavaMember.__call__(self, *args)
[2024-12-24T06:39:16.313+0000] {docker.py:73} INFO -    1316 command = proto.CALL_COMMAND_NAME +\
[2024-12-24T06:39:16.313+0000] {docker.py:73} INFO -    1317     self.command_header +\
[2024-12-24T06:39:16.314+0000] {docker.py:73} INFO -    1318     args_command +\
[2024-12-24T06:39:16.314+0000] {docker.py:73} INFO -    1319     proto.END_COMMAND_PART
[2024-12-24T06:39:16.314+0000] {docker.py:73} INFO -    1321 answer = self.gateway_client.send_command(command)
[2024-12-24T06:39:16.314+0000] {docker.py:73} INFO - -> 1322 return_value = get_return_value(
[2024-12-24T06:39:16.314+0000] {docker.py:73} INFO -    1323     answer, self.gateway_client, self.target_id, self.name)
[2024-12-24T06:39:16.314+0000] {docker.py:73} INFO -    1325 for temp_arg in temp_args:
[2024-12-24T06:39:16.315+0000] {docker.py:73} INFO -    1326     if hasattr(temp_arg, "_detach"):
[2024-12-24T06:39:16.315+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.315+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/pyspark/errors/exceptions/captured.py:179, in capture_sql_exception.<locals>.deco(*a, **kw)
[2024-12-24T06:39:16.315+0000] {docker.py:73} INFO -     177 def deco(*a: Any, **kw: Any) -> Any:
[2024-12-24T06:39:16.315+0000] {docker.py:73} INFO -     178     try:
[2024-12-24T06:39:16.315+0000] {docker.py:73} INFO - --> 179         return f(*a, **kw)
[2024-12-24T06:39:16.315+0000] {docker.py:73} INFO -     180     except Py4JJavaError as e:
[2024-12-24T06:39:16.316+0000] {docker.py:73} INFO -     181         converted = convert_exception(e.java_exception)
[2024-12-24T06:39:16.316+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.316+0000] {docker.py:73} INFO - File /opt/conda/lib/python3.11/site-packages/py4j/protocol.py:326, in get_return_value(answer, gateway_client, target_id, name)
[2024-12-24T06:39:16.316+0000] {docker.py:73} INFO -     324 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
[2024-12-24T06:39:16.316+0000] {docker.py:73} INFO -     325 if answer[1] == REFERENCE_TYPE:
[2024-12-24T06:39:16.316+0000] {docker.py:73} INFO - --> 326     raise Py4JJavaError(
[2024-12-24T06:39:16.317+0000] {docker.py:73} INFO -     327         "An error occurred while calling {0}{1}{2}.\n".
[2024-12-24T06:39:16.317+0000] {docker.py:73} INFO -     328         format(target_id, ".", name), value)
[2024-12-24T06:39:16.317+0000] {docker.py:73} INFO -     329 else:
[2024-12-24T06:39:16.317+0000] {docker.py:73} INFO -     330     raise Py4JError(
[2024-12-24T06:39:16.317+0000] {docker.py:73} INFO -     331         "An error occurred while calling {0}{1}{2}. Trace:\n{3}\n".
[2024-12-24T06:39:16.318+0000] {docker.py:73} INFO -     332         format(target_id, ".", name, value))
[2024-12-24T06:39:16.318+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.318+0000] {docker.py:73} INFO - Py4JJavaError: An error occurred while calling z:io.delta.tables.DeltaTable.isDeltaTable.
[2024-12-24T06:39:16.318+0000] {docker.py:73} INFO - : com.google.common.util.concurrent.ExecutionError: java.lang.NoClassDefFoundError: io/delta/storage/LogStore
[2024-12-24T06:39:16.318+0000] {docker.py:73} INFO - Please ensure that the delta-storage dependency is included.
[2024-12-24T06:39:16.318+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.319+0000] {docker.py:73} INFO - If using Python, please ensure you call `configure_spark_with_delta_pip` or use
[2024-12-24T06:39:16.319+0000] {docker.py:73} INFO - `--packages io.delta:delta-spark_<scala-version>:<delta-lake-version>`.
[2024-12-24T06:39:16.319+0000] {docker.py:73} INFO - See https://docs.delta.io/latest/quick-start.html#python.
[2024-12-24T06:39:16.319+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.319+0000] {docker.py:73} INFO - More information about this dependency and how to include it can be found here:
[2024-12-24T06:39:16.319+0000] {docker.py:73} INFO - https://docs.delta.io/latest/porting.html#delta-lake-1-1-or-below-to-delta-lake-1-2-or-above.
[2024-12-24T06:39:16.320+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.320+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2261)
[2024-12-24T06:39:16.320+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache.get(LocalCache.java:4000)
[2024-12-24T06:39:16.320+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4789)
[2024-12-24T06:39:16.320+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.getDeltaLogFromCache$1(DeltaLog.scala:790)
[2024-12-24T06:39:16.321+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.apply(DeltaLog.scala:800)
[2024-12-24T06:39:16.321+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.apply(DeltaLog.scala:710)
[2024-12-24T06:39:16.321+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.forTable(DeltaLog.scala:672)
[2024-12-24T06:39:16.321+0000] {docker.py:73} INFO - 	at io.delta.tables.DeltaTable$.isDeltaTable(DeltaTable.scala:811)
[2024-12-24T06:39:16.321+0000] {docker.py:73} INFO - 	at io.delta.tables.DeltaTable.isDeltaTable(DeltaTable.scala)
[2024-12-24T06:39:16.321+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-12-24T06:39:16.321+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-12-24T06:39:16.322+0000] {docker.py:73} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-12-24T06:39:16.322+0000] {docker.py:73} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-12-24T06:39:16.322+0000] {docker.py:73} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-12-24T06:39:16.322+0000] {docker.py:73} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-24T06:39:16.322+0000] {docker.py:73} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2024-12-24T06:39:16.323+0000] {docker.py:73} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-12-24T06:39:16.323+0000] {docker.py:73} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-12-24T06:39:16.323+0000] {docker.py:73} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-24T06:39:16.323+0000] {docker.py:73} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-24T06:39:16.323+0000] {docker.py:73} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-24T06:39:16.323+0000] {docker.py:73} INFO - Caused by: java.lang.NoClassDefFoundError: io/delta/storage/LogStore
[2024-12-24T06:39:16.324+0000] {docker.py:73} INFO - Please ensure that the delta-storage dependency is included.
[2024-12-24T06:39:16.324+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.324+0000] {docker.py:73} INFO - If using Python, please ensure you call `configure_spark_with_delta_pip` or use
[2024-12-24T06:39:16.324+0000] {docker.py:73} INFO - `--packages io.delta:delta-spark_<scala-version>:<delta-lake-version>`.
[2024-12-24T06:39:16.324+0000] {docker.py:73} INFO - See https://docs.delta.io/latest/quick-start.html#python.
[2024-12-24T06:39:16.324+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.325+0000] {docker.py:73} INFO - More information about this dependency and how to include it can be found here:
[2024-12-24T06:39:16.325+0000] {docker.py:73} INFO - https://docs.delta.io/latest/porting.html#delta-lake-1-1-or-below-to-delta-lake-1-2-or-above.
[2024-12-24T06:39:16.325+0000] {docker.py:73} INFO - 
[2024-12-24T06:39:16.325+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaErrorsBase.missingDeltaStorageJar(DeltaErrors.scala:2659)
[2024-12-24T06:39:16.325+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaErrorsBase.missingDeltaStorageJar$(DeltaErrors.scala:2656)
[2024-12-24T06:39:16.325+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaErrors$.missingDeltaStorageJar(DeltaErrors.scala:3203)
[2024-12-24T06:39:16.326+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.storage.DelegatingLogStore$.<init>(DelegatingLogStore.scala:163)
[2024-12-24T06:39:16.326+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.storage.DelegatingLogStore$.<clinit>(DelegatingLogStore.scala)
[2024-12-24T06:39:16.326+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.storage.DelegatingLogStore.$anonfun$schemeBasedLogStore$1(DelegatingLogStore.scala:67)
[2024-12-24T06:39:16.326+0000] {docker.py:73} INFO - 	at scala.Option.orElse(Option.scala:447)
[2024-12-24T06:39:16.326+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.storage.DelegatingLogStore.schemeBasedLogStore(DelegatingLogStore.scala:67)
[2024-12-24T06:39:16.326+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.storage.DelegatingLogStore.getDelegate(DelegatingLogStore.scala:85)
[2024-12-24T06:39:16.327+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.storage.DelegatingLogStore.read(DelegatingLogStore.scala:96)
[2024-12-24T06:39:16.327+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.Checkpoints.$anonfun$loadMetadataFromFile$1(Checkpoints.scala:361)
[2024-12-24T06:39:16.327+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[2024-12-24T06:39:16.327+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[2024-12-24T06:39:16.327+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog.recordFrameProfile(DeltaLog.scala:72)
[2024-12-24T06:39:16.327+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[2024-12-24T06:39:16.327+0000] {docker.py:73} INFO - 	at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[2024-12-24T06:39:16.328+0000] {docker.py:73} INFO - 	at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[2024-12-24T06:39:16.328+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog.recordOperation(DeltaLog.scala:72)
[2024-12-24T06:39:16.328+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[2024-12-24T06:39:16.328+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[2024-12-24T06:39:16.328+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[2024-12-24T06:39:16.329+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog.recordDeltaOperation(DeltaLog.scala:72)
[2024-12-24T06:39:16.329+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.Checkpoints.loadMetadataFromFile(Checkpoints.scala:360)
[2024-12-24T06:39:16.329+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.Checkpoints.readLastCheckpointFile(Checkpoints.scala:354)
[2024-12-24T06:39:16.329+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.Checkpoints.readLastCheckpointFile$(Checkpoints.scala:353)
[2024-12-24T06:39:16.329+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog.readLastCheckpointFile(DeltaLog.scala:72)
[2024-12-24T06:39:16.329+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$getSnapshotAtInit$1(SnapshotManagement.scala:436)
[2024-12-24T06:39:16.330+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[2024-12-24T06:39:16.330+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[2024-12-24T06:39:16.330+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog.recordFrameProfile(DeltaLog.scala:72)
[2024-12-24T06:39:16.330+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.SnapshotManagement.getSnapshotAtInit(SnapshotManagement.scala:434)
[2024-12-24T06:39:16.330+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.SnapshotManagement.getSnapshotAtInit$(SnapshotManagement.scala:433)
[2024-12-24T06:39:16.331+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog.getSnapshotAtInit(DeltaLog.scala:72)
[2024-12-24T06:39:16.331+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.SnapshotManagement.$init$(SnapshotManagement.scala:60)
[2024-12-24T06:39:16.331+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog.<init>(DeltaLog.scala:78)
[2024-12-24T06:39:16.331+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$4(DeltaLog.scala:779)
[2024-12-24T06:39:16.332+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[2024-12-24T06:39:16.332+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$3(DeltaLog.scala:774)
[2024-12-24T06:39:16.332+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[2024-12-24T06:39:16.332+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[2024-12-24T06:39:16.332+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.recordFrameProfile(DeltaLog.scala:598)
[2024-12-24T06:39:16.332+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[2024-12-24T06:39:16.333+0000] {docker.py:73} INFO - 	at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[2024-12-24T06:39:16.333+0000] {docker.py:73} INFO - 	at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[2024-12-24T06:39:16.333+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.recordOperation(DeltaLog.scala:598)
[2024-12-24T06:39:16.333+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[2024-12-24T06:39:16.333+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[2024-12-24T06:39:16.333+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[2024-12-24T06:39:16.334+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.recordDeltaOperation(DeltaLog.scala:598)
[2024-12-24T06:39:16.334+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.createDeltaLog$1(DeltaLog.scala:773)
[2024-12-24T06:39:16.334+0000] {docker.py:73} INFO - 	at org.apache.spark.sql.delta.DeltaLog$.$anonfun$apply$5(DeltaLog.scala:791)
[2024-12-24T06:39:16.334+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4792)
[2024-12-24T06:39:16.334+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3599)
[2024-12-24T06:39:16.334+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2379)
[2024-12-24T06:39:16.335+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2342)
[2024-12-24T06:39:16.335+0000] {docker.py:73} INFO - 	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2257)
[2024-12-24T06:39:16.335+0000] {docker.py:73} INFO - 	... 20 more
[2024-12-24T06:39:16.888+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 519, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 396, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 453, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-12-24T06:39:16.921+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=dag_Calendar_NTBK, task_id=NB_Notebook_job, run_id=manual__2024-12-24T06:37:51.454895+00:00, execution_date=20241224T063751, start_date=20241224T063753, end_date=20241224T063916
[2024-12-24T06:39:16.949+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-24T06:39:16.951+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 358 for task NB_Notebook_job (Docker container failed: {'StatusCode': 1}; 1192)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 519, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 396, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 453, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-12-24T06:39:17.007+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-24T06:39:17.078+0000] {taskinstance.py:3895} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-12-24T06:39:17.084+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
